src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/DeviceDetails/NewAzureStorSimpleNetworkConfig.cs

```
@@ -0,0 +1,222 @@
+            }
+            if(!TrySetIPv4Address(IPv6Gateway, out ipv6Gateway, "IPv6Gateway"))
+            {
+                return false;
+            }
+
+            // Only EnableCloud, Controller0 and controller1 IP Addresses can be set on Data0
+            if (InterfaceAlias == NetInterfaceId.Data0.ToString())
+            {
+                if(IPv4Address != null && IPv4Gateway != null && IPv4Netmask != null
+                    && IPv6Gateway != null && IPv6Prefix != null && EnableCloud != null)
+                {
+                    WriteVerbose(Resources.NetworkConfigData0AllowedSettings);
+                    WriteObject(null);
+                    return false;
+                }
+            }
+            // On other interfaces, Controller0 and Controller1 IP Addresses cannot be set
+            else
+            {
+                if (Controller0IPv4Address != null && Controller1IPv4Address != null)
+                {
+                    WriteVerbose(Resources.NetworkConfigControllerIPsNotAllowedOnOthers);
+                    WriteObject(null);
+                    return false;
+                }
+            }
+
+            // Controller0 and Controller1 IP Address cant be set on non Data0 interfaces
+
+            return true;
+        }
+
+        /// <summary>
+        /// Try to parse an IP Address from the provided string
+        /// </summary>
+        /// <param name="data">IP Address string</param>
+        /// <param name="ipAddress"></param>
+        /// <param name="paramName">Name of the param which is being processed (to be used for errors)</param>
+        private bool TrySetIPv4Address(string data, out IPAddress ipAddress, string paramName){
+            if(data == null){
+                ipAddress = null;
+                return true;
+            }
+            try{
+                ipAddress = IPAddress.Parse(data);
+                return true;
+            }
+            catch(FormatException){
```

does it only throw FormatException? Or should you catch all exceptions here?